packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -217,13 +217,15 @@ export class VueElement extends BaseClass {
     const resolve = (def: InnerComponentDef) => {
       const { props, styles } = def
       const hasOptions = !isArray(props)
-      const rawKeys = props ? (hasOptions ? Object.keys(props) : props) : []
+      const propsVal = props ? (hasOptions ? Object.keys(props) : props) : []
+      const attrs = Object.keys(this._props).filter( p => !propsVal.includes(p))
+      const rawKeys = [...propsVal.concat(attrs)]
 
       // cast Number-type props set before resolve
       let numberProps
       if (hasOptions) {
         for (const key in this._props) {
-          const opt = props[key]
+          const opt = props && props[key]
```

Comment:
This in-loop check would not be necessary with the change about setting `hasOptions` that I commented about above.